mvc/src/main/java/nextstep/mvc/controller/tobe/scanner/ControllerScanner.java

```diff
@@ -0,0 +1,39 @@
+package nextstep.mvc.controller.tobe.scanner;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import nextstep.web.annotation.Controller;
+import org.reflections.Reflections;
+
+public class ControllerScanner {
+
+    private final Reflections reflections;
+
+    public ControllerScanner(Reflections reflections) {
+        this.reflections = reflections;
+    }
+
+    public Map<Class<?>, Object> getControllers() {
+        Map<Class<?>, Object> controllers = new HashMap<>();
+        Set<Class<?>> annotatedWithController = reflections.getTypesAnnotatedWith(Controller.class);
+
+        annotatedWithController.forEach(controller ->
+                controllers.put(controller, createInstance(controller))
+        );
+
+        return controllers;
```

Comment:
```suggestion
        return annotatedWithController.stream()
                .collect(Collectors.toMap(
                        controller -> controller,
                        this::createInstance
                ));
```
스트림을 쓸 수도 있겠네요!!! 그냥 참고만 부탁드려요~